backend/de.metas.material/dispo-commons/src/main/java/de/metas/material/dispo/commons/repository/CandidateRepositoryWriteService.java

```diff
@@ -721,13 +719,20 @@ public DeleteResult deleteCandidateById(@NonNull final CandidateId candidateId)
 	{
 		final I_MD_Candidate candidateRecord = load(candidateId, I_MD_Candidate.class);
 		final DeleteResult deleteResult = new DeleteResult(candidateId,
-														   DateAndSeqNo.builder()
-																   .date(TimeUtil.asInstantNonNull(candidateRecord.getDateProjected()))
-																   .seqNo(candidateRecord.getSeqNo())
-																   .build(),
-														   candidateRecord.getQty());
-
-		deleteRecord(candidateRecord);
+				DateAndSeqNo.builder()
+						.date(TimeUtil.asInstantNonNull(candidateRecord.getDateProjected()))
+						.seqNo(candidateRecord.getSeqNo())
+						.build(),
+				candidateRecord.getQty());
+		deleteRelatedRecordsForId(candidateRecord, new HashSet<>());
```

Comment:
pls introduce deleteRelatedRecordsForId(candidateRecord)

Reply:
deleteRelatedRecordsForCandidate(candidateRecord) of course, because we're passing a candidate, not an Id.